src/index.ts

```diff
@@ -1,29 +1,30 @@
-import { Telegraf, Markup } from 'telegraf';
+import { Telegraf, Markup, Context } from 'telegraf';
 
 import { VercelRequest, VercelResponse } from '@vercel/node';
 import { development, production } from './core';
 
-import { createClient } from '@supabase/supabase-js'
-import { Database } from './supabase/database.types';
-import { EBotUserState, IBotUser } from './interfaces/bot-users';
-import { adaptCtx2User } from './lib/utils';
-
-
+import { IBotUser } from './interfaces/bot-users';
+import {
+  adaptCtx2User,
+  createUserByTelegram,
+  getUserByEmail,
+  getUserByTelegram,
+  handleCreateUser,
+  initEmailAuthentication,
+  verifyEmail,
+} from './lib/utils';
 
 /*
   SETUP
 */
 
-// Create a single supabase client for interacting with your database
-const supabaseUrl = process.env.SUPABASE_URL!
-const supabaseKey = process.env.SUPABASE_KEY!
-
-const supabase = createClient<Database>(supabaseUrl, supabaseKey)
-// console.log(supabase)
-
 const BOT_TOKEN = process.env.BOT_TOKEN || '';
 const ENVIRONMENT = process.env.NODE_ENV || '';
 
+let user: any;
```

Comment:
I don't think you can use variables like this. In production, the same server instance might not respond to subsequent requests. We should probably use Vercel KV https://vercel.com/docs/storage/vercel-kv or Supabase for storing variables on a per user basis.

Reply:
Yes, correct! Redis would be a great option here to manage these variables.